lib/api/objectPut.js

```diff
@@ -58,31 +64,24 @@ function _storeIt(bucket, objectKey, objMD, authInfo, canonicalID, cipherBundle,
         headers['x-amz-acl'] = request.query['x-amz-acl'];
     }
     const metadataStoreParams = {
-        objectKey,
-        authInfo,
-        metaHeaders,
-        size,
+        objectKey, authInfo, metaHeaders, size,
         contentType: request.headers['content-type'],
         cacheControl: request.headers['cache-control'],
         contentDisposition: request.headers['content-disposition'],
-        contentEncoding:
-            removeAWSChunked(request.headers['content-encoding']),
-        expires: request.headers.expires,
-        headers,
-        log,
+        contentEncoding: removeAWSChunked(request.headers['content-encoding']),
+        expires: request.headers.expires, headers, log,
+        isDeleteMarker: request.isDeleteMarker,
```

Comment:
changed createAndStoreObject to accept a new parameter 'isDeleteMarker' so we don't have to modify the request object in delete / multiobjectdelete apis